From a820dd37a8fd4bc95bb3502ce2219254f78d7adc Mon Sep 17 00:00:00 2001
From: Kyle Harrison <khwebmail@gmail.com>
Date: Sun, 19 Dec 2021 16:49:04 +0000
Subject: [PATCH] hwc: Update dependencies for S

ld.lld: error: undefined symbol: SkFILEWStream::SkFILEWStream(char const*)
ld.lld: error: undefined symbol: SkEncodeImage(SkWStream*, SkPixmap const&, SkEncodedImageFormat, int)
ld.lld: error: undefined symbol: SkFILEWStream::~SkFILEWStream()

- Builing against libhwui no longer fulfils the skia dependency.
- Link against the smaller libskia_renderengine library which does not require all of the codec dependencies of the full libskia.

Change-Id: Ib77f379859c6af9acf787b98fdd848959a54c69f
---
 libhwcomposer/Android.mk | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/libhwcomposer/Android.mk b/libhwcomposer/Android.mk
index 8f1c29e10a..3bd29c5781 100644
--- a/libhwcomposer/Android.mk
+++ b/libhwcomposer/Android.mk
@@ -23,7 +23,8 @@ LOCAL_CFLAGS                  := $(common_flags) -DLOG_TAG=\"qdhwcomposer\" -Wno
                                  -Wno-float-conversion -Wno-unused-parameter
 
 ifeq ($(TARGET_USES_QCOM_BSP),true)
-LOCAL_SHARED_LIBRARIES += libhwui
+LOCAL_STATIC_LIBRARIES += libskia_renderengine
+LOCAL_SHARED_LIBRARIES += libnativewindow libGLESv2 libpng
 ifeq ($(GET_FRAMEBUFFER_FORMAT_FROM_HWC),true)
     LOCAL_CFLAGS += -DGET_FRAMEBUFFER_FORMAT_FROM_HWC
 endif
-- 
2.34.1

